### PR TITLE
support async-await state flags

### PR DESCRIPTION
**Pull request by jamestalmage**
support async-await state flags

Adds support for monitoring the `async` and `generator` flags set by espower.

See power-assert-js/espower#23

### CODE REVIEW

File: lib/capturable.js [hunk 1]
```diff
@@ -11,16 +11,23 @@ module.exports = function capturable () {
...
+            content: args.content,
+            filepath: args.filepath,
+            line: args.line
+        };
+        if (args.generator) {
+            source.generator = true;
+        }
+        if (args.async) {
+            source.async = true;
+        }
```
Comment (jamestalmage) [inline lib/capturable.js]: @twada 
Is source the correct place to put these?
Or should they go on powerAssertContext?
I don't understand what the purpose of the two different objects are.

Comment (twada) [inline lib/capturable.js]: @jamestalmage Correct.

` _expr` function captures and collects values for each assertion argument.
So when assertion expression is `assert.equal(foo, bar)`, `_expr` is called twice (foo and bar).

- `source` is for information of assertion expression code itself, so `source` of two args are the same.
- `powerAssertContext` is for captured values and its paths, so `powerAssertContext` of two args are not same.

I agree that these variable names are confusing. I'll fix it when this pull-req lands.

Comment (jamestalmage) [inline lib/capturable.js]: OK,
So they should stay on source like I have them?

**Comment by jamestalmage** [comment]
@twada Tests are passing when I run them on travis:
https://travis-ci.org/jamestalmage/empower/builds/89599931

Failures here seem to be network related (travis having some issues?)

**Comment by jamestalmage** [comment]
@twada Only thing left is to bump espower to point at new version once you publish

File: lib/capturable.js (hunk 1 above)
Comment (twada) [inline lib/capturable.js]: Yes.

**Comment by twada** [comment]
@jamestalmage Just released [espower 1.2.1](https://github.com/power-assert-js/espower/blob/master/CHANGELOG.md#121-2015-11-06).

**Comment by jamestalmage** [comment]
@twada ready for merge and release here

**Comment by twada** [comment]
@jamestalmage I'm on a train so wait an hour please. :train:

**Comment by twada** [comment]
@jamestalmage Okay, I'm going to release this.

**Comment by twada** [comment]
Just released [empower 1.1.0](https://github.com/power-assert-js/empower/blob/master/CHANGELOG.md#110-2015-11-07) :tada:

**Pull request merged**